### PR TITLE
Automate Multiple Tag Subscription

### LINKED ISSUE #4759: Detect subscriptions in signup `return_to` param and correctly subscribe new users in one step
Currently, people can be directed to the signup form with the following format:

https://stable.publiclab.org/signup?return_to=/subscribe/multiple/tag/arduino,games (for example)

That means they should get signed up in that form (and it works the same in the modal signup), and then upon making an account, get redirected to `/subscribe/multiple/tag/arduino,games`, which will subscribe them to the 2 tags `arduino` and `games` and then AGAIN redirect them to `/subscriptions`. 

But this doesn't happen - you just get directed to the /dashboard page and you see the standard welcome page, though you do get shown an option (in the long welcome alert) to "pick up where you left off" with a link:

https://github.com/publiclab/plots2/blob/a3918357e87f01149aaf35251adc6899e0ee2c84/app/controllers/users_controller.rb#L28-L32

Let's think about how we want people to be signed up. New users shouldn't be dropped off on the subscriptions page; that makes sense. 

Maybe we should... detect the `return_to=/subscribe/multiple/tag/arduino,games` format, or the `params[:return_to]` and parse it to see if there are subscriptions in there? 

Could we match `/subscribe/multiple/tag/arduino,games` and actually parse out the tagnames to subscribe them automatically? 

Other ideas? What's best here?

### PR DESCRIPTION
**Pull request by SidharthBansal**
Fixes #4759 (<=== Add issue number here)
image
image

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by SidharthBansal** [comment]
@gauravano @jywarren this is ready. can you please review it?

**Comment by SidharthBansal** [comment]
As this pr is over 100 lines of code, I will provide the tests in the next pr. As per @jywarren 's suggestions prs should be less than 50 lines of code. The behaviour can be checked on the unstable branch.

**Comment by SidharthBansal** [comment]
@jywarren I have pushed the tests. I think this is complete now.
Thanks

**Comment by jywarren** [comment]
This is awesome. I really really want to protect this from future disruption with a test. What do you think?

**Comment by jywarren** [comment]
Oh whoa. sorry your last comment hadn't loaded. Checking!!!

File: test/integration/login_flow_test.rb [hunk 1]
```diff
@@ -30,6 +30,30 @@ class LoginFlowTest < ActionDispatch::IntegrationTest
...
+            statement2: I18n.t('spamaway.human.statement2'),
+            statement3: I18n.t('spamaway.human.statement3'),
+            statement4: I18n.t('spamaway.human.statement4')
+          },
+          return_to: '/subscribe/multiple/tag/arduino,games'
+        }
+    assert_response :redirect
+    # a success here would mean sent back to form with errors
+    assert_redirected_to '/dashboard'
+    assert_equal 'Registration successful. Welcome to our community!You are now following \'arduino,games\'.',flash[:notice]
```
Comment (jywarren) [inline test/integration/login_flow_test.rb]: This is great. Can we add a space before `You are`?

**Comment by jywarren** [comment]
Awesome!!! Great test, too. What do you think of an integration test where we actually start at `/subscribe/multiple/tag/arduino,games` in a logged-out state, and confirm the entire pathway? I really want to protect this well.

**Comment by jywarren** [comment]
Oh wow! Is that tested here? https://github.com/publiclab/plots2/pull/5566/files#diff-640ca8f656ee6c17b73ddd38ec0caeecR107

**Comment by jywarren** [comment]
Wait, so are people left on the dashboard still at the end, or the subscriptions page? Thank you!!!

**Comment by SidharthBansal** [comment]
> Oh wow! Is that tested here? https://github.com/publiclab/plots2/pull/5566/files#diff-640ca8f656ee6c17b73ddd38ec0caeecR107

Yeah I wrote the test. Thanks

**Pull request merged**